### PR TITLE
AB#87732 Summary of Delivery - First version of calculation

### PR DESCRIPTION
**Pull request by Carq**
image

### CODE REVIEW

File: src/HE.Investment.AHP.Contract/Delivery/Queries/GetDeliveryPhaseDetailsQuery.cs [hunk 1]
```diff
@@ -3,4 +3,4 @@
 
 namespace HE.Investment.AHP.Contract.Delivery.Queries;
 
-public record GetDeliveryPhaseDetailsQuery(AhpApplicationId ApplicationId, DeliveryPhaseId DeliveryPhaseId) : IRequest<DeliveryPhaseDetails>;
+public record GetDeliveryPhaseDetailsQuery(AhpApplicationId ApplicationId, DeliveryPhaseId DeliveryPhaseId, bool CalculateSummary = false) : IRequest<DeliveryPhaseDetails>;
```
Comment (Antek012) [inline src/HE.Investment.AHP.Contract/Delivery/Queries/GetDeliveryPhaseDetailsQuery.cs:6]: Maybe IncludeSummary?

File: src/HE.Investment.AHP.Domain.Tests/Delivery/Entities/TestDataBuilders/DeliveryPhaseEntityBuilder.cs [hunk 2]
```diff
@@ -112,6 +112,13 @@ public DeliveryPhaseEntityBuilder WithReconfiguringExisting()
         return this;
     }
 
+    public DeliveryPhaseEntityBuilder WithBuildActivity(BuildActivityType buildActivityType)
```
Comment (Antek012) [inline src/HE.Investment.AHP.Domain.Tests/Delivery/Entities/TestDataBuilders/DeliveryPhaseEntityBuilder.cs:115]: WithRehabBuildActivity as you provide buildActivityType only

File: src/HE.Investment.AHP.Domain/Delivery/Entities/DeliveryPhaseEntity.cs [hunk 3]
```diff
@@ -137,11 +141,14 @@ public void ProvideAdditionalPaymentRequest(IsAdditionalPaymentRequested? isAddi
...
     }
 
+
+
     public void Complete()
     {
         if (!IsAnswered())
         {
-            throw new DomainValidationException(ValidationErrorMessage.SectionIsNotCompleted);
+            // TODO #67047: throw and handle exception
```
Comment (Antek012) [inline src/HE.Investment.AHP.Domain/Delivery/Entities/DeliveryPhaseEntity.cs:150]: please revert

File: src/HE.Investment.AHP.Domain/Delivery/Entities/DeliveryPhaseEntity.cs [hunk 4]
```diff
@@ -177,22 +205,13 @@ public bool IsReconfiguringExistingNeeded()
 
     private bool IsAnswered()
     {
-        var reconfigureExistingValid = !IsReconfiguringExistingNeeded() || ReconfiguringExisting.HasValue;
```
Comment (Antek012) [inline src/HE.Investment.AHP.Domain/Delivery/Entities/DeliveryPhaseEntity.cs:180]: Why this was removed? merge problem?

File: src/HE.Investment.AHP.Domain/Delivery/ValueObjects/MilestoneFramework.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
+using Dawn;
```
Comment (Antek012) [inline src/HE.Investment.AHP.Domain/Delivery/ValueObjects/MilestoneFramework.cs:1]: https://www.nuget.org/packages/Ardalis.GuardClauses

**Review by Antek012 — APPROVED** [review]

**Review by dawid-pawulski — APPROVED** [review]

**Pull request merged**